src/_common/content/content-editor/modals/gif/gif-modal.vue

```diff
@@ -243,6 +261,25 @@ export default class AppContentEditorGifModal extends mixins(BaseModal) {
 		this.loadingCategories = true;
 		this.populateCategories();
 	}
+
+	async toggleFavorite(searchResult: SearchResult) {
+		const toggleTo = !searchResult.favorite;
+		searchResult.favorite = toggleTo;
+
+		const url =
+			`/web/content/tenor/` +
+			(toggleTo ? 'add-favorite' : 'remove-favorite') +
+			`/${searchResult.id}`;
+		const response = await Api.sendRequest(url, {});
+		if (!response.success) {
+			showErrorGrowl(
+				$gettext(
+					`You've reached the maximum amount of favorite GIFs. Unfavorite some from your Favorites list to add more.`
```

Comment:
Let's use this message: "Sorry, you have reached the maximum number of favorite GIFs. To add more, please remove some from your Favorites list."